filament/src/RenderPass.cpp

```diff
@@ -581,15 +591,15 @@ void RenderPass::Executor::recordDriverCommands(backend::DriverApi& driver,
                         CONFIG_MAX_BONE_COUNT * sizeof(PerRenderableUibBone));
             }
 
-            auto morphTargetBuffer = info.morphTargetBuffer;
-            if (UTILS_UNLIKELY(morphTargetBuffer)) {
+            if (UTILS_UNLIKELY(info.morphWeightBuffer)) {
                 // Instead of using a UBO per primitive, we could also have a single UBO for all primitives
                 // and use bindUniformBufferRange which might be more efficient.
-                auto morphing = soaMorphing[info.index];
-                assert_invariant(morphing.handle);
-                driver.bindUniformBuffer(BindingPoints::PER_RENDERABLE_MORPHING, morphing.handle);
-                assert_invariant(morphing.count <= morphTargetBuffer->getCount());
-                morphTargetBuffer->bind(driver);
+                driver.bindUniformBuffer(BindingPoints::PER_RENDERABLE_MORPHING,
+                        info.morphWeightBuffer);
+                if (UTILS_UNLIKELY(info.morphTargetBuffer)) {
```

Comment:
Add a comment describing the conditions in which `morphWeightBuffer` is true, but `morphTargetBuffer` is false.  I think this can occur when skinning is enabled but morphing is not, right?

Reply:
Yes it's exactly.